frontend/src/shared/components/CreatePassword.tsx

```diff
@@ -0,0 +1,117 @@
+        setOpened(false);
+    });
+
+    const handlePaste = async (e: React.MouseEvent, propName: string) => {
+        e.preventDefault();
+        userPassForm.setFieldValue(propName, await paste());
+    };
+
+    const pasteButton = (propName: string) => (
+        <Button
+            bg={modalButtonColor}
+            size="sm"
+            p="5px"
+            onClick={(e: React.MouseEvent) => handlePaste(e, propName)}
+        >
+            <IconClipboard />
+        </Button>
+    );
+
+    return (
+        <>
+            <Modal
+                centered={true}
+                opened={opened}
+                onClose={() => setOpened(false)}
+                title="Add New Password"
+            >
+                <form onSubmit={handlePasswordSave}>
+                    <TextInput
+                        m={modalTextInputMargin}
+                        placeholder="Website"
+                        label="Website"
+                        withAsterisk
+                        required
+                        {...userPassForm.getInputProps('website')}
+                        rightSection={pasteButton('website')}
+                    />
+                    <TextInput
+                        m={modalTextInputMargin}
+                        placeholder="Username"
+                        label="Username"
+                        withAsterisk
+                        required
+                        {...userPassForm.getInputProps('username')}
+                        rightSection={pasteButton('username')}
+                    />
+                    <TextInput
+                        m={modalTextInputMargin}
+                        placeholder="Password"
+                        label="Password"
+                        withAsterisk
+                        required
+                        {...userPassForm.getInputProps('password')}
+                        rightSection={pasteButton('password')}
+                    />
+                    <Group position="center">
+                        <Button
+                            bg={modalButtonColor}
+                            m={modalButtonMargin}
+                            size={modalButtonSize}
+                            type="submit"
+                        >
+                            Save
+                        </Button>
+                        <Button
+                            bg={modalButtonColor}
+                            m={modalButtonMargin}
+                            size={modalButtonSize}
+                            type="reset"
+                            onClick={() => setOpened(false)}
+                        >
+                            Cancel
+                        </Button>
+                    </Group>
+                </form>
+            </Modal>
+            <Group position="right">
+                <Button bg={modalButtonColor} m={modalButtonMargin} onClick={handleModalOpen}>
+                    New
```

Comment:
We probably want to use the top bar for these buttons. One way to trigger them using react-router-dom. Will discuss this. Commenting for future reference.